elide-core/src/main/java/com/yahoo/elide/core/filter/FilterPredicate.java

```diff
@@ -5,110 +5,109 @@
     }
 
-    public FilterPredicate(Path.PathElement pathElement, Operator op, List<Object> values) {
-        this(new Path(Arrays.asList(pathElement)), op, values);
-    }
-
-    public FilterPredicate(Path.PathElement pathElement, Operator op) {
-        this(new Path(Arrays.asList(pathElement)), op, Collections.emptyList());
+    public FilterPredicate(PathElement pathElement, Operator op, List<Object> values) {
+        this(new Path(Collections.singletonList(pathElement)), op, values);
     }
 
-    public FilterPredicate(Path path, Operator op) {
-        this(path, op, Collections.emptyList());
+    public FilterPredicate(FilterPredicate copy) {
+        this(copy.path, copy.operator, copy.values);
     }
 
     public FilterPredicate(Path path, Operator op, List<Object> values) {
-        this.path = path;
         this.operator = op;
-        this.values = values;
-    }
-
-    public FilterPredicate(FilterPredicate copy) {
-        this.path = new Path(new ArrayList<>(copy.getPath().getPathElements()));
-        this.operator = copy.getOperator();
-        this.values = new ArrayList<>(copy.getValues());
-    }
-
-    public String getField() {
-        List<Path.PathElement> elements = path.getPathElements();
-        Path.PathElement last = elements.get(elements.size() - 1);
-        return last.getFieldName();
-    }
-
-    public String getFieldPath() {
-        StringBuilder fieldPath = new StringBuilder();
-        for (Path.PathElement pathElement : path.getPathElements()) {
-            if (fieldPath.length() != 0) {
-                fieldPath.append(PERIOD);
-            }
-            fieldPath.append(pathElement.getFieldName());
-        }
-        return fieldPath.toString();
+        this.path = new Path(path);
+        this.values = ImmutableList.copyOf(values);
+        this.field = path.lastElement()
+                .map(PathElement::getFieldName)
+                .orElse(null);
+        this.fieldPath = path.getPathElements().stream()
+                .map(PathElement::getFieldName)
+                .collect(Collectors.joining(PERIOD));
     }
 
     /**
-     * Get a unique name for this predicate to be used as a parameter name.
-     * @return unique name
+     * Compute the parameter value/name pairings.
+     * @return the filter parameters for this predicate
      */
-    public String getParameterName() {
-        return getFieldPath().replace(PERIOD, UNDERSCORE) + UNDERSCORE + Integer.toHexString(hashCode());
+    public List<FilterParameter> getParameters() {
+        String baseName = String.format("%s_%s_",
+                getFieldPath().replace(PERIOD, UNDERSCORE),
+                Integer.toHexString(hashCode()));
+        return IntStream.range(0, values.size())
+                .mapToObj(idx -> new FilterParameter(String.format("%s%d", baseName, idx), values.get(idx)))
+                .collect(Collectors.toList());
+    }
+
+    public FilterPredicate scopedBy(PathElement scope) {
```

Comment:
Maybe a comment here about what this is doing?  Could this be package and not public?

Reply:
Not the same package, so can't change the access level. I'll add a comment though.